sdk/src/main/java/com/uid2/EUIDManager.kt

```diff
@@ -0,0 +1,122 @@
+package com.uid2
+
+import android.content.Context
+import com.uid2.UID2Manager.Companion.APPLICATION_ID_DEFAULT
+import com.uid2.network.DefaultNetworkSession
+import com.uid2.network.NetworkSession
+import com.uid2.storage.FileStorageManager
+import com.uid2.storage.FileStorageManager.Store.EUID
+import com.uid2.storage.StorageManager
+import com.uid2.utils.InputUtils
+import com.uid2.utils.Logger
+import com.uid2.utils.TimeUtils
+import kotlinx.coroutines.Dispatchers
+
+public class EUIDManager {
```

Comment:
The existence of this different class that's just a factory for a slightly different UID2Manager seems pretty weird. From my lack of knowledge, what is exactly changing here from the normal UIDManager.getInstance ? 

We could either go the factory function route, the extension route, the initializing lambda route, etc. But to be able to give a helpful suggestion I'd need to know what is changing exactly here. 

FWIW The java-ish UID2Manager shape is not exactly helpful either here XD

Reply:
Another alternative is, if possible, working toward this being an object instead of a class because you could pretty much instantiate an `EUIDManager` class here that'd be kind of useless (and misleading).

Reply:
The challenge here is trying to keep backwards compatability with existing integrators.

>Another alternative is, if possible, working toward this being an object instead of a class because you could pretty much instantiate an EUIDManager class here that'd be kind of useless (and misleading).

Dave and I have spoken about this previously, and would both love to move away from a singleton. Our current thinking is to consider this when moving to 2.0.0 of the SDK, so later - as it impacts a lot of other things (e.g. all the documentation already created for publishers).

I do agree we should consider what's easiest for publishers to consume. Since a lot will support an app that supports both UID2 and EUID, the existing interface means they will be required to do this constantly:

```
if (isUID2) { UID2Manager.getInstant() } else { EUID.getInstance() }
```

The alternative would be to just change the UID2Manager.init method to allow UID2 vs EUID configuration (and environment) and everyone access directly via UID2Manager.